crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1564,168 +1554,144 @@ where
+        // if the lowest buffered ancestor is invalid
+        if self
+            .check_invalid_ancestor_with_head(
+                lowest_buffered_ancestor,
+                sync_target_state.head_block_hash,
+            )
+            .is_some()
+        {
+            warn!(
+                target: "consensus::engine",
+                invalid_ancestor = %lowest_buffered_ancestor,
+                head = %sync_target_state.head_block_hash,
+                "Current head has an invalid ancestor"
+            );
+            return Ok(())
+        }
 
-                let sync_target_state = match self.forkchoice_state_tracker.sync_target_state() {
-                    Some(current_state) => current_state,
-                    None => {
-                        // This is only possible if the node was run with `debug.tip`
-                        // argument and without CL.
-                        warn!(target: "consensus::engine", "No fork choice state available");
-                        return None
-                    }
-                };
+        // get the block number of the finalized block, if we have it
+        let newest_finalized = self
+            .blockchain
+            .buffered_header_by_hash(sync_target_state.finalized_block_hash)
+            .map(|header| header.number);
 
-                // Next, we check if we need to schedule another pipeline run or transition
-                // to live sync via tree.
-                // This can arise if we buffer the forkchoice head, and if the head is an
-                // ancestor of an invalid block.
-                //
-                //  * The forkchoice head could be buffered if it were first sent as a `newPayload`
-                //    request.
-                //
-                // In this case, we won't have the head hash in the database, so we would
-                // set the pipeline sync target to a known-invalid head.
-                //
-                // This is why we check the invalid header cache here.
-                let lowest_buffered_ancestor =
-                    self.lowest_buffered_ancestor_or(sync_target_state.head_block_hash);
-
-                // this inserts the head if the lowest buffered ancestor is invalid
-                if self
-                    .check_invalid_ancestor_with_head(
-                        lowest_buffered_ancestor,
-                        sync_target_state.head_block_hash,
-                    )
-                    .is_none()
-                {
-                    // get the block number of the finalized block, if we have it
-                    let newest_finalized = self
-                        .forkchoice_state_tracker
-                        .sync_target_state()
-                        .map(|s| s.finalized_block_hash)
-                        .and_then(|h| self.blockchain.buffered_header_by_hash(h))
-                        .map(|header| header.number);
```

Comment:
this unnecessarily retrieved optional `self.forkchoice_state_tracker.sync_target_state()`, but we already retrieved and ensured that it's some up above